.github/workflows/makefile.yml

```diff
@@ -15,31 +26,84 @@ jobs:
-      run: |
-        sudo echo 'debconf debconf/frontend select Noninteractive' | sudo debconf-set-selections
-        sudo apt-get -y install apt-utils
+    - name: Set up QEMU        
+      uses: docker/setup-qemu-action@v1
 
-    - name: Install Docker CE
-      run: |
-        sudo apt-get install ca-certificates curl gnupg lsb-release
-        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
-        echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
-        sudo apt-get update
-        sudo apt-get install docker-ce docker-ce-cli containerd.io
+    - name: Set up Docker Buildx        
+      uses: docker/setup-buildx-action@v1
+
+#    - name: Login to DockerHub
+#      if: github.event_name != 'pull_request'
+#      uses: docker/login-action@v1 
+#      with:
+#        username: ${{ secrets.DOCKERHUB_USERNAME }}
+#        password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-    - name: Container Build
+    - name: Login to GitHub Container Registry
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1 
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Create tags and labels
+      id: image_meta
+      uses: docker/metadata-action@v3
+      with:
+        sep-tags: ","
+        sep-labels: ","
+        # list of names to use as base name for tags
+        images: |
+          ipdk-io/ipdk-${{ matrix.os }}
+          ghcr.io/${{ github.repository_owner }}/ipdk-${{ matrix.os }}
+        # Docker tags based on the following events/attributes
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha
+
+    - name: Install IPDK cli
       run: |
         cd build
         ./ipdk install ${{ matrix.os }}
-        ipdk build --no-cache
+ 
+    - name: IPDK image build and push
+      if: github.event_name != 'pull_request'
+      run: |
+        ipdk build --no-cache \
+          --platform $PLATFORM \
+          --tags ${{ steps.image_meta.outputs.tags }} \
+          --labels ${{ steps.image_meta.outputs.labels }} \
+          --push
+
+    - name: IPDK image build and export
+      if: github.event_name == 'pull_request'
+      run: |
+        ipdk build --no-cache \
+          --platform $PLATFORM \
+          --tags ${{ steps.image_meta.outputs.tags }} \
+          --labels ${{ steps.image_meta.outputs.labels }} \
+          --export /tmp/${{ matrix.os }}.tar
+
+    - name: Upload build artifact
+      if: github.event_name == 'pull_request'
```

Comment:
I am not sure if this is a good idea. Uploading Fedora 33 (1.06 GB) + Ubuntu 20.04 (862 MB) artifacts for a few days, every time a pull request has been created or updated, might result in the error message: [`Cannot upload artifacts. You may have reached your capacity.`](https://github.community/t/reached-artifact-limit/16860)

Reply:
Thanks for mentioning this limit possibility!!!

I have checked the docs and everywhere it is stated that the (artifact) storage limit is for private repositories only (and all the people mentioning the problem are talking about private repositories). The other limit that I have found is api calls (total calls in x time) but we don't have that problem here as it is only one file (and as such a very limited number of calls). Also repeated updated pr's within the runtime if the previous pr push are canceled an as such not stored. 

But you are right, storing for more then one day is not needed so I will add `retention-days: 1` so the artifacts will be purged after 1 day.

Reply:
@stolsma Thanks for looking into this. Specifying `retention-days` sounds good to me and I think you are right. I checked with the GitHub virtual support and the answer was:

> Actions usage is free for public repositories and self-hosted runners. For private repositories, each account receives a certain amount of free minutes and storage.

@mestery What do you think?

Reply:
One thing we could do is only push images on merges, which would limit how much storage we use and stop pushing images which we build as part of pull request testing. Another thing we could do is set a retention policy (see [here](https://github.com/actions/upload-artifact/issues/59#issuecomment-1026859091) for things in a reasonable manner.

Reply:
I think 1 day retention is fine, and only pushing on merge is also the right move, thanks for taking care of that!